Lib/http/server.py

```diff
@@ -1192,20 +1192,42 @@ def run_cgi(self):
                 nbytes = int(length)
             except (TypeError, ValueError):
                 nbytes = 0
+
+            data = None
+            rfile = None
+
+            if self.command.lower() == "post" and nbytes > 0:
+                data = self.rfile.read(nbytes)
+                if len(data) < nbytes:
+                    import tempfile
+                    rfile = tempfile.TemporaryFile("wb+")
```

Comment:
Rather than an actual file, could we use `BytesIO` or (at worst), `mmap` here instead?

Reply:
Good idea. All we care about is reading from socket and temporarily storing to a different file like object (which can support large files). I will try with BytesIO.